File: Koordinat/soal2_250051_250093_250012.java
/*
Nama Program : Program Koordinat
Anggota      : - Amela Dzakiah Majid (140810250051)
               - Atika Shafira (140810250093)
               - Syifa Dwi Amirah (140810250012)
Tanggal Buat : 24 September 2026
Deskripsi    : Membuat program mencari titik tengah dengan passing object
*/

import java.util.Scanner;

class Koordinat {
    private double absis;
    private double ordinat;

    public Koordinat() {
        absis = 0;
        ordinat = 0;
    }

    public Koordinat(double absis, double ordinat) {
        this.absis = absis;
        this.ordinat = ordinat;
    }

    public void setKoordinat(double absis, double ordinat) {
        this.absis = absis;
        this.ordinat = ordinat;
    }

    public void setAbsis(double absis) {
        this.absis = absis;
    }

    public void setOrdinat(double ordinat) {
        this.ordinat = ordinat;
    }

    public double getAbsis() {
        return absis;
    }

    public double getOrdinat() {
        return ordinat;
    }

    public void inputKoordinat(Scanner input) {
        System.out.print("Masukkan absis: ");
        absis = Double.parseDouble(input.nextLine());

        System.out.print("Masukkan ordinat: ");
        ordinat = Double.parseDouble(input.nextLine());
    }

    public void printKoordinat() {
        System.out.println("Nilai Absis   = " + absis);
        System.out.println("Nilai Ordinat = " + ordinat);
    }

    public void titikTengahVoid(Koordinat P1, Koordinat P2) {
        this.absis = (P1.absis + P2.absis) / 2;
        this.ordinat = (P1.ordinat + P2.ordinat) / 2;
    }

    public Koordinat titikTengahReturn(Koordinat P) {
        Koordinat hasil = new Koordinat();

        hasil.absis = (P.absis + this.absis) / 2;
        hasil.ordinat = (P.ordinat + this.ordinat) / 2;

        return hasil;
    }
}


public class Koordinat {
    static Scanner input = new Scanner(System.in);
    public static void menu() {
        Koordinat titikA = new Koordinat(2, 1);
        Koordinat titikB = new Koordinat(6, 3);
        Koordinat titikT = new Koordinat();

        int pilihan = 0;
        while (pilihan != 4) {
            System.out.println("\n===== MENU PASSING KOORDINAT =====");
            System.out.println("1. Titik A dipanggil ke Titik B");
            System.out.println("2. Titik B dipanggil ke Titik A");
            System.out.println("3. Titik A dan B dipanggil ke Titik T");
            System.out.println("4. Keluar");
            System.out.print("Pilih menu: ");

            pilihan = Integer.parseInt(input.nextLine());

            if (pilihan == 1) {

                System.out.println("\nTitik A");
                titikA.printKoordinat();

                System.out.println("\nTitik B");
                titikB.printKoordinat();

                System.out.println("\nTitik A dipanggil ke Titik B");

                titikT = titikB.titikTengahReturn(titikA);

                System.out.println("\nTitik T");
                titikT.printKoordinat();

            } else if (pilihan == 2) {

                System.out.println("\nTitik A");
                titikA.printKoordinat();

                System.out.println("\nTitik B");
                titikB.printKoordinat();

                System.out.println("\nTitik B dipanggil ke Titik A");

                titikT = titikA.titikTengahReturn(titikB);

                System.out.println("\nTitik T");
                titikT.printKoordinat();

            } else if (pilihan == 3) {

                System.out.println("\nTitik A");
                titikA.printKoordinat();

                System.out.println("\nTitik B");
                titikB.printKoordinat();

                System.out.println("\nTitik A dan Titik B dipanggil ke Titik T");

                titikT.titikTengahVoid(titikA, titikB);

                System.out.println("\nTitik T");
                titikT.printKoordinat();

            } else if (pilihan == 4) {

                System.out.println("Program selesai.");

            } else {

                System.out.println("Menu tidak tersedia!");
            }
        }
    }


    public static void main(String[] args) {
        menu();
        input.close();
    }
}

File: Koordinat/soal2_250051_250093_250012.py
"""
Nama Program : Program Koordinat
Anggota      : - Amela Dzakiah Majid (140810250051)
               - Atika Shafira (140810250093)
               - Syifa Dwi Amirah (140810250012)
Tanggal Buat : 24 September 2026
Deskripsi    : Membuat program mencari titik tengah dengan passing object
"""
class Koordinat:
    def __init__(self, absis=0, ordinat=0):
        self.absis = absis
        self.ordinat = ordinat

    def setKoordinat(self, absis, ordinat):
        self.absis = absis
        self.ordinat = ordinat

    def setAbsis(self, absis):
        self.absis = absis

    def setOrdinat(self, ordinat):
        self.ordinat = ordinat

    def getAbsis(self):
        return self.absis

    def getOrdinat(self):
        return self.ordinat

    def inputKoordinat(self):
        self.absis = float(input("Masukkan absis: "))
        self.ordinat = float(input("Masukkan ordinat: "))

    def printKoordinat(self):
        print("Nilai Absis   =", self.absis)
        print("Nilai Ordinat =", self.ordinat)

    def titikTengahVoid(self, P1, P2):
        self.absis = (P1.absis + P2.absis) / 2
        self.ordinat = (P1.ordinat + P2.ordinat) / 2

    def titikTengahReturn(self, P):
        hasil = Koordinat()
        hasil.absis = (P.absis + self.absis) / 2
        hasil.ordinat = (P.ordinat + self.ordinat) / 2
        return hasil


def menu():
    titikA = Koordinat(2, 1)
    titikB = Koordinat(6, 3)
    titikT = Koordinat()

    pilihan = 0

    while pilihan != 4:
        print("\n===== MENU PASSING KOORDINAT =====")
        print("1. Titik A dipanggil ke Titik B")
        print("2. Titik B dipanggil ke Titik A")
        print("3. Titik A dan B dipanggil ke Titik T")
        print("4. Keluar")

        pilihan = int(input("Pilih menu: "))

        if pilihan == 1:
            print("\nTitik A")
            titikA.printKoordinat()

            print("\nTitik B")
            titikB.printKoordinat()

            print("\nTitik A dipanggil ke Titik B")
            titikT = titikB.titikTengahReturn(titikA)

            print("\nTitik T")
            titikT.printKoordinat()

        elif pilihan == 2:
            print("\nTitik A")
            titikA.printKoordinat()

            print("\nTitik B")
            titikB.printKoordinat()

            print("\nTitik B dipanggil ke Titik A")
            titikT = titikA.titikTengahReturn(titikB)

            print("\nTitik T")
            titikT.printKoordinat()

        elif pilihan == 3:
            print("\nTitik A")
            titikA.printKoordinat()

            print("\nTitik B")
            titikB.printKoordinat()

            print("\nTitik A dan Titik B dipanggil ke Titik T")
            titikT.titikTengahVoid(titikA, titikB)

            print("\nTitik T")
            titikT.printKoordinat()

        elif pilihan == 4:
            print("Program selesai.")

        else:
            print("Menu tidak tersedia!")


def main():
    menu()


if __name__ == "__main__":
    main()